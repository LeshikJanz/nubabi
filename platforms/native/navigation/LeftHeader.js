// @flow
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { View, StyleSheet } from 'react-native';
import { delay } from 'lodash';
import { NubabiIcon } from '../components';
import { HEADER_FONT_COLOR } from '../../../core/themes/defaultTheme';

type Props = {
  navigate: () => void,
};

class LeftHeader extends Component {
  props: Props;

  static contextTypes = {
    setActiveTransition: PropTypes.func,
    getActiveTransition: PropTypes.func,
  };

  navigateTo = (routeName: string) => {
    this.context.setActiveTransition('chooseBaby', () => {
      delay(this.props.navigate.bind(null, routeName), 0.5);
    });
  };

  render() {
    return (
      <View>
        <NubabiIcon
          name="changeBabyIcon"
          style={styles.headerIcon}
          onPress={() => this.navigateTo('chooseBaby')}
        />
      </View>
    );
  }
}
const styles = StyleSheet.create({
  headerIcon: {
    padding: 10,
    backgroundColor: 'transparent',
    color: HEADER_FONT_COLOR,
    fontSize: 17,
  },
});

export default LeftHeader;
