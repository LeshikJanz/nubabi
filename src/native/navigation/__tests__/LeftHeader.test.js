import 'react-native';
import React from 'react';
import LeftHeader from '../LeftHeader';
import renderer from 'react-test-renderer';

test('it renders correctly', () => {
  const tree = renderer.create(<LeftHeader />).toJSON();

  expect(tree).toMatchSnapshot();
});
