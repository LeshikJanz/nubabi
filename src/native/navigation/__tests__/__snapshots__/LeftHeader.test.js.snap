// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    onPress={[Function]}
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 12,
        },
        Object {
          "backgroundColor": "transparent",
          "color": "#3D414B",
          "fontSize": 17,
          "padding": 10,
        },
        Object {
          "fontFamily": "nubabi",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    Ó§Å
  </Text>
</View>
`;
